sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/EncodingPerFileTest.java

```diff
@@ -51,40 +50,43 @@ public void should_treat_as_match_when_roslyn_entry_missing_for_file() throws IO
     Charset roslynCharset = StandardCharsets.UTF_8;
     Charset sqCharset = StandardCharsets.UTF_16;
 
-    HashMap<URI, Charset> encodingPerUri = new HashMap<>();
-    encodingPerUri.put(fileUri, roslynCharset);
-    assertEncodingMatch(encodingPerUri, URI.create("dummy"), sqCharset, true);
+    assertEncodingMatch(roslynCharset, URI.create("dummy"), sqCharset, true);
```

Comment:
I don't understand the name of this test - `when_roslyn_entry_missing_for_file`

but `encodingPerUri.put(fileUri, roslynCharset);` does contain the roslyn entry, right?

Reply:
Be careful about this little difference:
```
encodingPerUri.put(this.fileUri.toString(), roslynCharset);  // Test does this
encodingPerUri.put(fileUri.toString(), roslynCharset);       // You've mentioned something like this
```
The orchestration of the test adds `this.fileUri`. That's result of `temp.newFile().toURI()`. Entry for `temp.newFile().toURI()` is added with declared encoding. 

And in this test, entry for `URI.create("dummy")` is tested. This entry does not exist, as the test name says.

I think the name of the test describes it's content well.